src/components/ProcessDocumentScreen/ProcessDocumentContent/ProcessDocumentContent.tsx

```diff
@@ -0,0 +1,150 @@
+import { generateFileName } from "../../../utils";
+import { ErrorCard } from "../../UI/ErrorCard";
+import { ProcessedDocumentTag } from "../ProcessedDocumentTag";
+
+interface ProcessDocumentContentProps {
+  network: Network;
+  queueState: QueueState;
+  queueType: QueueType;
+  failedDocuments: WrappedDocument[];
+  pendingDocuments: WrappedDocument[];
+  successDocuments: WrappedDocument[];
+
+  fileName: string;
+
+  downloadErrorName: string;
+  downloadErrorLink: string;
+
+  downloadAllFn: MouseEventHandler<HTMLButtonElement>;
+
+  processAnotherDocumentFn: MouseEventHandler<HTMLButtonElement>;
+}
+
+export const ProcessDocumentContent: FunctionComponent<ProcessDocumentContentProps> = ({
+  network,
+
+  queueState,
+  queueType,
+  failedDocuments,
+  pendingDocuments,
+  successDocuments,
+
+  fileName,
+
+  downloadErrorName,
+  downloadErrorLink,
+
+  downloadAllFn,
+
+  processAnotherDocumentFn,
+}) => {
+  const isIssuingFlow = queueType === QueueType.ISSUE;
+  const isNetworkError = queueState === QueueState.ERROR && failedDocuments.length === 0;
+  const isDocumentError = queueState !== QueueState.ERROR && failedDocuments.length > 0;
+  const isDocumentSuccess =
+    queueState !== QueueState.ERROR && (pendingDocuments.length > 0 || successDocuments.length > 0);
+  const isAllFailed =
+    queueState === QueueState.CONFIRMED && failedDocuments.length > 0 && successDocuments.length === 0;
+
+  const ErrorLogButton = () => {
+    return (
+      <Button className="bg-rose flex mx-auto border-rose hover:bg-rose-400">
+        <a className="text-white hover:text-white" download={downloadErrorName} href={downloadErrorLink}>
+          Download Error Log
+        </a>
+      </Button>
+    );
+  };
+
+  const DownloadAllButton = () => {
+    return (
+      <Button
+        className="bg-white text-cerulean hover:bg-cloud-100 mb-4"
+        data-testid="download-all-button"
+        onClick={downloadAllFn}
+      >
+        <div className="flex">
+          <Download />
+          <div className="text-cerulean ml-2">Download all</div>
+        </div>
+      </Button>
+    );
+  };
+
+  return (
+    <div className="pb-6">
+      {isDocumentError && (
+        <>
+          <div className="mb-4 flex-grow py-3" data-testid="total-number-of-documents">
+            {failedDocuments.length} document(s) failed
+          </div>
+
+          <div className={`${successDocuments.length > 0 ? `pb-8 mb-8 border-b border-cloud-200` : ""}`}>
+            <ErrorCard
+              title={`The document(s) could not be ${
+                isIssuingFlow ? "issued" : "revoked"
+              } at this time. Please fix the errors and try again.`}
+              description={
+                <div>
+                  {failedDocuments.map((failedDocument, index) => {
+                    return (
+                      <div key={`${failedDocument.fileName}-${index}`}>
+                        {generateFileName({
```

Comment:
```suggestion
                      <div key={index}>
                        {generateFileName({
```

i think this might work fine enuff if im not wrong